### PR TITLE
Control mapping

### PR DESCRIPTION
**Pull request by gnawinganimal**
I thought it would be helpful to make some kind of control mapping utility. Originally it would have looked like:
```kotlin
val drive by mapping<ControlOption2, Double>(controlOption2) {
     control(ControlOption2.JOYSTICK) { joystick2.y }
     control(ControlOption2.CONTROLLER) { controller2.getY(GenericHID.Hand.kLeft) }
}
```

but as it turns out I was making it a lot more complicated. Here's what it looks like now:
```kotlin
val drive by when (controlOption2) {
    ControlOption2.JOYSTICK -> get { joystick2.y }
    ControlOption2.CONTROLLER -> get { controller2.getY(GenericHID.Hand.kLeft) }
}
```
It's just a `when` statement combined by a tiny utility function that this PR adds, `get`.

### CODE REVIEW

**Comment by gnawinganimal** [comment]
I shall just merge this now

**Comment by codingvic** [comment]
Hey Aiden how's it going?! Srry I saw this yesterday but forgot to comment on it b/c I couldn't really contribute anything... How are you guys doing? Is this code for the robot? I have SOOO many questions and I bet half of them would be answered if I just checked my slack... Also maybe delete this comment after you've read it b/c I know Greg likes having the comments be like code-only soooooooooooo... thanks for including me!

**Comment by codingvic** [comment]
This code just gets a controller and joystick input right?

**Comment by gnawinganimal** [comment]
Hey, how's it going? I'll explain it a little better.

The code actually doesn't do anything except add a function called `get`. What `get` does is just sets up a `ReadOnlyProperty` that always returns the result of the function placed in `get`.
For instance, if you were to do this:
```kotlin
val five by get { 5 }
```
you would just have a value that always returns `5`.
The use for this is that if you have a `var` that is already defined, like `joystick.y`, you can create a variable that always returns `joystick.y`.
```kotlin
val drive by get { joystick.y }
```
But if you have multiple options for controls, for instance, if the driver has the option for a joystick and a controller, you can combine this with a `when` statement and an enum:
```kotlin
val drive by when (selectedControls) {
    Controls.JOYSTICK -> get { joystick.y }
    Controls.CONTROLLER -> get { controller.getY(GenericHID.Hand.kLeft) }
}
```
##### TLDR;
`get` is a function that could be used for anything, but it helps with creating control schemes.

**Comment by gnawinganimal** [comment]
Also, don't worry about the comment, that's what they're for 👍

**Comment by mckelveygreg** [comment]
Ha, no this is great! I'm all for Github as a social network :)

**Comment by codingvic** [comment]
Oh ok that makes more sense, can I ask what the when function is used for? It's not like a loop is it?
